### PR TITLE
feat: 🎸 validate chain spec version when connecting

### PR DESCRIPTION
**Pull request by monitz87**
### Description

- Fetch and compare chain spec version as well as rpc node version with the expected ranges, throw if not compatible

### Breaking Changes

- Throw an error instead of a warning when the chain versions aren't
within the expected ranges

### JIRA Link

https://polymath.atlassian.net/browse/DA-324

### Checklist

- [ ] Updated the Readme.md (if required) ?

### CODE REVIEW

File: src/Polymesh.ts [hunk 1]
```diff
@@ -87,16 +87,7 @@ export class Polymesh {
...
-      if (
-        error instanceof PolymeshError &&
-        error.message.includes('Unsupported Polymesh version')
-      ) {
-        console.warn(error.message);
-      } else {
-        throw error;
-      }
-    });
+    await assertExpectedChainVersion(nodeUrl);
```
[the remark this answers is not in the document]
Comment (monitz87) [reply src/Polymesh.ts:90]: The confluence doc says it should be an error for major version mismatches, which I think is the sensible choice

File: src/utils/internal.ts [hunk 2]
```diff
@@ -1001,25 +1003,74 @@ export function assertExpectedChainVersion(nodeUrl: string): Promise<void> {
...
+
+        /*
+         * the spec version number comes as a single number (i.e. 5000000). It should be parsed as xxx_yyy_zzz
+         * where xxx is the major version, yyy is the minor version, and zzz is the patch version. So for example, 5001023
+         * would be version 5.1.23
+         */
+        const specVersionAsSemver = specVersion
+          .toString()
+          // add dot separator every three digits
+          .replace(/(\d)(?=(\d{3})+(?!\d))/g, '$1.')
```
[the remark this answers is not in the document]
Comment (monitz87) [reply src/utils/internal.ts:1050]: Still works, I tried it with spec version 50_000_000 for example

**Review by VictorVicente — CHANGES_REQUESTED** [review]

**Review by VictorVicente — APPROVED** [review]

**Pull request merged**